### PR TITLE
extra closure methods

### PR DESCRIPTION
**Pull request by dr7ana**
Close methods needed for lokinet integration

### CODE REVIEW

File: include/quic/endpoint.hpp [hunk 1]
```diff
@@ -171,6 +171,8 @@ namespace oxen::quic
 
         void close_connection(Connection& conn, int code = NGTCP2_NO_ERROR, std::string_view msg = "NO_ERROR"sv);
 
+        void close_connection(ConnectionID cid, int code = NGTCP2_NO_ERROR, std::string_view msg = "NO_ERROR"sv);
```
Comment (tewinget) [inline include/quic/endpoint.hpp:174]: It looks like `Connection::close_connection` is calling the previous declaration of this, so this appears unused?

Comment (dr7ana) [reply include/quic/endpoint.hpp:174]: It’s not unused, closing connections via SCID is useful for lokinet integration

Comment (tewinget) [reply include/quic/endpoint.hpp:174]: Ah okay, it's just unused in this commit; that makes more sense.

Comment (dr7ana) [reply include/quic/endpoint.hpp:174]: Oh i understand what you meant now! Yes exactly it’s for external use. Will PM you

**Review by tewinget — APPROVED** [review]

**Pull request merged**